packages/react-search-ui/src/containers/Facet.tsx

```diff
@@ -92,13 +92,30 @@ export class FacetContainer extends Component<
     if (!facetValues.length && !selectedValues.length) return null;
 
     if (searchTerm.trim()) {
-      facetValues = facetValues.filter((option) =>
-        typeof option.value === "string"
-          ? accentFold(option.value)
-              .toLowerCase()
-              .includes(accentFold(searchTerm).toLowerCase())
-          : false
-      );
+      facetValues = facetValues.filter((option) => {
+        let valueToSearch;
+        switch (typeof option.value) {
+          case "string":
+            valueToSearch = accentFold(option.value).toLowerCase();
+            break;
+          case "number":
+            valueToSearch = option.value.toString();
+            break;
+          case "object":
+            valueToSearch =
```

Comment:
nitpick: could you use optional chaining to improve readability? 

`typeof option?.value?.name === "string"`

Reply:
Done